src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -0,0 +1,307 @@
+		{
+			const matrix::Vector3f target_predicted_position = predict_future_x_ned_est(POSITION_FILTER_ALPHA, x_ned_est, v_ned_est,
+					a_ned_est);
+			_target_position_filtered.setParameters(_deltatime, POSITION_FILTER_ALPHA);
+			_target_position_filtered.update(target_predicted_position);
+		}
+
+		if (_param_nav_ft_fs.get() == FOLLOW_PERSPECTIVE_NONE) {
+			// NOTE: Switching between NONE any any other setting currently causes a jump in the setpoints
+			_offset_vector_filtered.reset(matrix::Vector2f{0, 0});
+
+		} else {
+			// Define and rotate offset vector based on follow-me perspective setting
+			const float new_follow_angle_deg = update_follow_me_angle_setting(_param_nav_ft_fs.get());
+
+			// Use shortest rotation to get to the new angle
+			// Example: if the current angle setting is 270, and the new angle setting is 0, it's
+			// faster to rotate to 360 rather than 0.
+			// Usually the controller would automatically take the shortest path. But here some trickery
+			// is necessary because the yaw angle is run through a low-pass filter.
+			if (_follow_angle_deg - new_follow_angle_deg  > 180.0f) {
+				_follow_angle_deg = new_follow_angle_deg + 360.0f;
+
+			} else if (_follow_angle_deg - new_follow_angle_deg  < -180.0f) {
+				_follow_angle_deg = new_follow_angle_deg - 360.0f;
+
+			} else {
+				_follow_angle_deg = new_follow_angle_deg;
+			}
+
+			// Lowpass the angle setting to smoothly transition to a new perspective when the user makes a change.
+			// In particular this has an effect when the setting is modified by 180 degrees, in which case the drone
+			// would pass above the target without the filter. The filtering makes it so that the drone flies around
+			// the target into the new postion.
+			_follow_angle_filtered.setParameters(_deltatime, FOLLOW_ANGLE_FILTER_ALPHA);
+			_follow_angle_filtered.update(_follow_angle_deg);
+
+			// Wrap around 360 degrees
+			if (_follow_angle_filtered.getState() > 360.0f) {
+				_follow_angle_filtered.reset(_follow_angle_filtered.getState() - 360.0f);
+				_follow_angle_deg = _follow_angle_deg - 360.0f;
+
+			} else if (_follow_angle_filtered.getState() < -360.0f) {
+				_follow_angle_filtered.reset(_follow_angle_filtered.getState() + 360.0f);
+				_follow_angle_deg = _follow_angle_deg + 360.0f;
+			}
+
+			// Assume the target's velocity vector is its heading and use it to construct the offset vector
+			// such that drone_pos_setpoint = target_pose + offset_vector
+			if (v_ned_est.norm() > MINIMUM_SPEED_FOR_HEADING_CHANGE &&
+			    v_ned_est.norm() > FLT_EPSILON) {
+				// Compute offset vector relative to target position. At the same time the offset vector defines the
+				// vieweing angle of the drone
+				_target_velocity_unit_vector = matrix::Vector2f(v_ned_est.xy()).unit_or_zero();
+			}
+
+			const double follow_angle_rad = (double)_follow_angle_filtered.getState() * M_DEG_TO_RAD;
```

Comment:
```suggestion
			const double follow_angle_rad = (double)math::radians(_follow_angle_filtered.getState());
```

Does this need to be double?

Reply:
As opposed to float you mean? No, absolutely not. It does not need the extra precision.